### ISSUE TITLE
Test failure: OutOfProcessToInProcessHostingModelSwitchWorks

### ISSUE THREAD

**Issue by aspnet-hello** [aspnet/IISIntegration#1349]
This test [fails](http://aspnetci/viewLog.html?buildId=541190&buildTypeId=XPlat_Windows_Win10Universe) occasionally with the following error:
```
'Could not find the assembly 'aspnetcorev2_inprocess.dll'' didn't match any event log messaged.
Expected: True
Actual:   False
   at Microsoft.AspNetCore.Server.IISIntegration.FunctionalTests.EventLogHelpers.VerifyEventLogEvent(IISDeploymentResult deploymentResult, ITestSink testSink, String expectedRegexMatchString) in /_/test/Common.FunctionalTests/Utilities/EventLogHelpers.cs:line 75
   at Microsoft.AspNetCore.Server.IISIntegration.FunctionalTests.ConfigurationChangeTests.OutOfProcessToInProcessHostingModelSwitchWorks() in /_/test/Common.FunctionalTests/ConfigurationChangeTests.cs:line 86
--- End of stack trace from previous location where exception was thrown ---
------- Stdout: -------
| [0.001s] TestLifetime Information: Starting test OutOfProcessToInProcessHostingModelSwitchWorks at 2018-08-28T21:45:38
| [0.005s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Information: Deploying [Variation] :: ServerType=IIS, Runtime=CoreClr, Arch=x64, BaseUrlHint=, Publish=True
| [0.008s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Debug: Processing C:\BuildAgent\temp\buildTmp\9c037664a2a14f60af14b80dc017c504\runtimes\unix\lib\netcoreapp2.0
| [0.009s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Debug:   Copying System.Security.Principal.Windows.dll
| [0.011s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Debug: Processing C:\BuildAgent\temp\buildTmp\9c037664a2a14f60af14b80dc017c504\runtimes\unix\lib
| [0.011s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Debug: Processing C:\BuildAgent\temp\buildTmp\9c037664a2a14f60af14b80dc017c504\runtimes\unix
| [0.013s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Debug: Processing C:\BuildAgent\temp\buildTmp\9c037664a2a14f60af14b80dc017c504\runtimes\win\lib\netcoreapp2.0
| [0.013s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Debug:   Copying System.Security.Principal.Windows.dll
| [0.016s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Debug: Processing C:\BuildAgent\temp\buildTmp\9c037664a2a14f60af14b80dc017c504\runtimes\win\lib\netcoreapp2.1
| [0.016s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Debug:   Copying System.Security.Cryptography.Cng.dll
| [0.021s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Debug: Processing C:\BuildAgent\temp\buildTmp\9c037664a2a14f60af14b80dc017c504\runtimes\win\lib
| [0.021s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Debug: Processing C:\BuildAgent\temp\buildTmp\9c037664a2a14f60af14b80dc017c504\runtimes\win
| [0.021s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Debug: Processing C:\BuildAgent\temp\buildTmp\9c037664a2a14f60af14b80dc017c504\runtimes
| [0.022s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Debug: Processing C:\BuildAgent\temp\buildTmp\9c037664a2a14f60af14b80dc017c504\wwwroot
| [0.022s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Debug:   Copying static.txt
| [0.023s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Debug: Processing C:\BuildAgent\temp\buildTmp\9c037664a2a14f60af14b80dc017c504
| [0.023s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Debug:   Copying Microsoft.AspNetCore.Authentication.Abstractions.dll
| [0.025s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Debug:   Copying Microsoft.AspNetCore.Authentication.Core.dll
| [0.026s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Debug:   Copying Microsoft.AspNetCore.Connections.Abstractions.dll
| [0.028s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Debug:   Copying Microsoft.AspNetCore.Hosting.Abstractions.dll
| [0.030s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Debug:   Copying Microsoft.AspNetCore.Hosting.dll
| [0.032s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Debug:   Copying Microsoft.AspNetCore.Hosting.Server.Abstractions.dll
| [0.034s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Debug:   Copying Microsoft.AspNetCore.Http.Abstractions.dll
| [0.036s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Debug:   Copying Microsoft.AspNetCore.Http.dll
| [0.038s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Debug:   Copying Microsoft.AspNetCore.Http.Extensions.dll
| [0.040s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Debug:   Copying Microsoft.AspNetCore.Http.Features.dll
| [0.043s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Debug:   Copying Microsoft.AspNetCore.HttpOverrides.dll
| [0.045s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Debug:   Copying Microsoft.AspNetCore.Server.IISIntegration.dll
| [0.047s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Debug:   Copying Microsoft.AspNetCore.Server.IISIntegration.pdb
| [0.048s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Debug:   Copying Microsoft.AspNetCore.Server.IISIntegration.xml
| [0.059s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Debug:   Copying Microsoft.AspNetCore.Server.Kestrel.Core.dll
| [0.061s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Debug:   Copying Microsoft.AspNetCore.Server.Kestrel.dll
| [0.063s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Debug:   Copying Microsoft.AspNetCore.Server.Kestrel.Https.dll
| [0.064s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Debug:   Copying Microsoft.AspNetCore.Server.Kestrel.Transport.Abstractions.dll
| [0.066s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Debug:   Copying Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.dll
| [0.068s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Debug:   Copying Microsoft.AspNetCore.WebUtilities.dll
| [0.069s] Microsoft.AspNetCore.Server.I
...
```
Other tests within that build may have failed with a similar message, but they are not listed here. Check the link above for more info.

This test failed on master.

CC @muratg

This issue was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by aspnet-hello** [comment]
Please use this workflow to address this flaky test issue, including checking applicable checkboxes and filling in the applicable "TODO" entries:

* Is this actually a flaky test?
  * No, this is a regular test failure, fix the test/product (TODO: Link to commit/PR)
  * Yes, proceed below...

* Is this test failure caused by product code flakiness? (Either this product, or another product this test depends on.)
  * [ ] File a bug against the product (TODO: Link to other bug)
  * Is it possible to change the test to avoid the flakiness?
    * Yes? Go to the "Change the test!" section.
    * No?
      * [ ] Disable the test (TODO: Link to PR/commit)
      * [ ] Wait for other bug to be resolved
      * [ ] Wait for us to get build that has the fix
      * [ ] Re-enable our test (TODO: Link to PR/commit)
      * [ ] Close this bug

* Is it that the test itself is flaky? This includes external transient problems (e.g. remote server problems, file system race condition, etc.)
  * Is there is a way to change our test to avoid this flakiness?
    * Yes? Change the test!
      * [ ] Change the test to avoid the flakiness, for example by using a different test strategy, or by adding retries w/ timeouts (TODO: Link to PR/commit)
      * [ ] Run the test 100 times locally as a sanity check.
      * [ ] Close this bug
    * No?
      * Is there any logging or extra information that we could add to make this more diagnosable when it happens again?
        * Yes?
            * [ ] Add the logging (TODO: Link to PR/commit)
        * No?
            * [ ] Delete the test because flaky tests are not useful (TODO: Link to PR/commit)

This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by muratg** [comment]
@jkotalik @pakrym please take a look.

**Comment by jkotalik** [comment]
Should be resolved by #1338, but will leave open for a few days.
